server/container/guice/guice-common/src/test/java/org/apache/james/MailsShouldBeWellReceived.java

```diff
@@ -62,9 +78,56 @@
     String SENDER = "bob@apache.org";
     String UTF_8_BODY = "UTF-8 character 'Ð'";
 
-    @Disabled("JAMES-1824 actually the body is changed in to 'UTF-8 character '?''")
+
+    static Message readFirstMessageJavax(int imapPort) throws MessagingException {
+        Session imapSession = Session.getDefaultInstance(new Properties());
+        Store store = imapSession.getStore("imap");
+        store.connect("localhost", imapPort, JAMES_USER, PASSWORD);
+        Folder inbox = store.getFolder(IMAPMessageReader.INBOX);
+        inbox.open(Folder.READ_ONLY);
+
+        CALMLY_AWAIT.untilAsserted(() ->
+            assertThat(searchForUnSeen(inbox))
+                .hasSize(1));
+
+        return Stream.of(searchForUnSeen(inbox))
+            .sorted(Comparator.comparing(Throwing.function(Message::getReceivedDate)))
+            .findFirst()
+            .get();
+    }
+
+    static Message[] searchForUnSeen(Folder inbox) throws MessagingException {
+        return inbox.search(new FlagTerm(new Flags(Flags.Flag.SEEN), false));
+    }
+
+    static void sendMessageJavax(GuiceJamesServer server) throws MessagingException {
+        Port smtpPort = server.getProbe(SmtpGuiceProbe.class).getSmtpPort();
+
+        Properties props = new Properties();
+        props.put("mail.smtp.host", "localhost");
+        props.put("mail.smtp.port", smtpPort.getValue() + "");
+        props.put("mail.smtp.auth", "true");
+        props.put("mail.smtp.allow8bitmime", "true"); //force to use 8bit(UTF-8), otherwise, it automatically converts into 7bit
+
+        Authenticator auth = new Authenticator() {
+            protected PasswordAuthentication getPasswordAuthentication() {
+                return new PasswordAuthentication(SENDER, PASSWORD);
+            }
+        };
+
+        Session session = Session.getInstance(props, auth);
+        Message message = new MimeMessage(session);
+        message.setFrom(new InternetAddress(SENDER));
+        message.setRecipients(Message.RecipientType.TO, InternetAddress.parse(JAMES_USER));
+        message.setText(UTF_8_BODY);
+
+        Transport.send(message);
+    }
+
+    @Disabled("Suspected JAMES-1824 actually the body is changed in to 'UTF-8 character '?''" +
+        "But finally seems like SMTPMessageSender & IMAPMessageReader don't treat UTF-8 character so well")
     @Test
-    default void mailsContentWithUtf8CharactersShouldBeKeptUnChanged(GuiceJamesServer server) throws Exception {
+    default void mailsContentWithUtf8CharactersShouldBeKeptUnChangedWithCurrentTestingTools(GuiceJamesServer server) throws Exception {
```

Comment:
testing tool tests should not be located on a James server generic contract IMO.

Shouldn't such a test be part of the SMTPMessageSender test suite?

Reply:
I'm on my way to move this test back to `SMTPMessageSenderTest`, and it's not the final stage, just want to show people James still work fine under UTF-8 bodies

Reply:
are you sure (I did not check) if UTF-8 bodies are supported by the RFC? 8-bit is very different than UTF-8 and should not be confused.

Reply:
> are you sure (I did not check) if UTF-8 bodies are supported by the RFC

I do not know too

But there's a popular use case where I want to send a mail containing Unicode characters (like `Đ` or `Ê`). And the mail client should perform it well, and I don't care about how the client works, but I expect that I can read the exactly mail but not the mail with Unicode characters rewritten in `?` question mark characters

Reply:
If you send an email containing Unicode characters, your mail client will convert the body, probably using a 7-bit encoding (base64 for example, or quoted printable).

Then the smtp server does not need to handle directly unicode characters.

Here you are checking that the smtp server is supporting utf-8. I'm not sure it is needed.

Reply:
> your mail client will convert the body, probably using a 7-bit encoding (base64 for example, or quoted printable)

Exactly this point, the James `SMTPMessageSender` doesn't do it properly, so the message I received at James side is in wrong content. I could remove these test about James. But the fault of the `SMTPMessageSender` still remains and I want to demonstrate it in the second commit.

Reply:
SMTPMessageSender is sending raw lines at the SMTP level, so I'm not very surprised that it does not handle this correctly. Can you check if SMTP is supposed to handle UTF-8?

BTW SMTPMessageSender is supposed to be a pretty low smtp interface, so isn't it expected?

Reply:
I found it, that is the underlying apache common net `SMTPClient` is using ISO-8.... charset by default. 

An another thing, do you agree to move this test to some where else? Like the James memory test, I want to close the JAMES-1824 but I need to have a proof